blivet/udev.py

```diff
@@ -87,16 +88,23 @@ def get_devices(subsystem="block"):
 def settle(quiet=False):
     """ Wait for the udev queue to settle.
 
-        :keyword bool quiet: bypass :meth:`blivet.util.run_program`
+        :keyword bool quiet: bypass :meth:`blivet.util.capture_output`
     """
     # wait maximal 300 seconds for udev to be done running blkid, lvm,
     # mdadm etc. This large timeout is needed when running on machines with
     # lots of disks, or with slow disks
+
     argv = ["udevadm", "settle", "--timeout=300"]
     if quiet:
-        subprocess.call(argv, close_fds=True)
+        proc = subprocess.run(argv, close_fds=True, capture_output=True, text=True, check=True)
+        out = proc.stderr
     else:
-        util.run_program(argv)
+        out = util.capture_output(argv, stderr_to_stdout=True)
+    if out.find('ignoring'):
```

Comment:
I was going to ask if we could do something else than parsing the output to check for the error message, like checking the return code, but `udevadm settle` just return 0 in this case. Maybe we could do some chroot detection ourselves?

Reply:
For reference:
Per udev logic it bypasses **trigger**, **settle**, **control** and **monitor** commands in chroot, and [chroot detection](https://github.com/systemd/systemd/blob/4e5c39597add4c0e3c479fc11d562f6ea92f22de/src/basic/virt.c#L861) is done by checking SYSTEMD_IGNORE_CHROOT env var and comparing whether **/proc/1/root** and **/** are the same directory.
We can copy this code for our purposes to bypass output parsing.
However, as of now udevadm prints the same message for all bypass scenarios, and it might be 'cleaner' to leave chroot detection to udevadm and analyse output instead.

Reply:
I would like to avoid parsing of the error message. In case the message ever changes, the whole issue will reappear.

I think that we should adapt referenced udev logic. Ideally with a comment describing what and why it is being done. Also it might be actually better for us that it covers only this scenario. Generic check may silently catch something we do not want to be caught.

Reply:
@japokorn do we want to rely on SYSTEMD_IGNORE_CHROOT or only pid 1 root dir?